core/src/main/scala/cromwell/core/callcaching/docker/registryv2/flows/HttpFlowWithRetry.scala

```diff
@@ -1,30 +1,53 @@
 package cromwell.core.callcaching.docker.registryv2.flows
 
 import akka.NotUsed
+import akka.actor.Scheduler
 import akka.http.scaladsl.model.{HttpRequest, HttpResponse, StatusCodes}
-import akka.stream.scaladsl.{Flow, GraphDSL, Merge, Partition}
-import akka.stream.{FanOutShape2, OverflowStrategy}
+import akka.stream.FanOutShape2
+import akka.stream.javadsl.MergePreferred
+import akka.stream.scaladsl.{Flow, GraphDSL, Partition}
 import cromwell.core.callcaching.docker.registryv2.flows.FlowUtils._
 import cromwell.core.callcaching.docker.registryv2.flows.HttpFlowWithRetry._
+import cromwell.core.retry.{Backoff, SimpleExponentialBackoff}
 
+import scala.concurrent.duration._
+import scala.concurrent.{ExecutionContext, Future}
+import scala.language.postfixOps
 import scala.util.Try
 
 object HttpFlowWithRetry {
-  def defaultIsRetryable(response: HttpResponse) = {
+  def isRetryable(response: HttpResponse) = {
     response.status match {
-      case StatusCodes.TooManyRequests => true
       case StatusCodes.InternalServerError => true
+      case StatusCodes.ServiceUnavailable => true
+      case StatusCodes.BadGateway => true
+      case StatusCodes.GatewayTimeout => true
+      case StatusCodes.RequestTimeout => true
+      case other => isTransient(response)
+    }
+  }
+
+  def isTransient(response: HttpResponse) = {
+    response.status match {
+      case StatusCodes.TooManyRequests => true
       case _ => false
     }
   }
 
+  def defaultRequestBackoff(): Backoff = SimpleExponentialBackoff(1 second, 2 minutes, 3D)
+  
   /**
     * In order to allow for retries, the http context object needs to encapsulate the original request,
     * so that it can be re-submitted if necessary.
     * This type provides this encapsulation by creating a pair of (T, HttpRequest) where T is the custom type that
     * the user wishes to use as a context for their specific use case.
     */
-  type ContextWithRequest[T] = (T, HttpRequest)
+  case class ContextWithRequest[T](userContext: T, request: HttpRequest, currentAttempt: Int, backoff: Backoff) {
+    // Backoff is mutable ! This will change every time !
```

Comment:
ToL: `backoff` is two words. This is one of our cromwellian pervasive errors, just like `testkit`.